### PR TITLE
Update docs for experimental modules

### LINKED ISSUE #4386: Akka HTTP and Play Streams docs
Go over the docs and make sure they're up to date for the 2.4.0 release.

* https://www.playframework.com/documentation/2.4.x/AkkaHttpServer
* https://www.playframework.com/documentation/2.4.x/ReactiveStreamsIntegration

### PR DESCRIPTION
**Pull request by richdougherty**
Resolves #4386.

### CODE REVIEW

File: documentation/manual/experimental/AkkaHttpServer.md [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+Play 2's main server is built on top of [Netty](http://netty.io/). In Play 2.4 we started experimenting with an experimental server based on [Akka HTTP](http://doc.akka.io/docs/akka-stream-and-http-experimental/current/). Akka HTTP is an HTTP library built on to of Akka. It is written by the authors of [Spray](http://spray.io/).
 
-The experimental Akka HTTP backend is a technical proof of concept. It is not intended for production use and it doesn't implement the full Play API. The purpose of the new backend is to trial Akka HTTP as a possible backend for a future version of Play. The backend also serves as a valuable test case for our friends on the Akka project.
+The purpose of this backend is:
+
+* to check that the Akka HTTP API provides all the features that Play needs
+* to gain knowledge about Akka HTTP in case we want to use it in Play in the future.
+
+The Akka HTTP is mostly a proof of concept. We do **not** recommend that you use it for anything other than learning about Play or Akka HTTP server code.
```
Comment (marcospereira) [inline documentation/manual/experimental/AkkaHttpServer.md]: I got in doubt here: is Akka HTTP itself a proof of concept or its use in Play 2.4?

File: documentation/manual/experimental/ReactiveStreamsIntegration.md [hunk 2]
```diff
@@ -3,18 +3,25 @@
 
 > **Play experimental libraries are not ready for production use**. APIs may change. Features may not work properly.
 
-[Reactive Streams](http://www.reactive-streams.org/) is a specification and SPI that is currently under development. Reactive Streams provides a common standard that allows different stream implementations to be connected together. The SPI is quite small, with just a few simple interfaces such as `Publisher` and `Subscriber`.
+[Reactive Streams](http://www.reactive-streams.org/) is a new standard that gives a common API for asynchronous streams. Play 2.4 introduces some wrappers to convert Play's [[Iteratees and Enumerators|Iteratees]] into Reactive Streams objects. This means that Play can integrate with other software that supports Reactive Streams, e.g. [Akka](http://doc.akka.io/docs/akka-stream-and-http-experimental/current/), [RxJava](https://github.com/ReactiveX/RxJavaReactiveStreams) and [others](http://www.reactive-streams.org/announce-1.0.0#implementations).
```
Comment (marcospereira) [inline documentation/manual/experimental/ReactiveStreamsIntegration.md]: `[Akka Streams](...)`

**Comment by richdougherty** [comment]
@marcospereira, I tried to answer your question about the Akka HTTP module by updating the docs. Let me know if it's still confusing.

**Comment by richdougherty** [comment]
@play-pull-request-validator, please try again.

@jroper, a resolution error, hopefully transient:
```
[info] Loading project definition from /scratch/jenkins/workspace/play-master-PRs/framework/project
[warn] 	module not found: org.scalariform#scalariform_2.10;0.1.5-20140822-69e2e30
[warn] ==== typesafe-ivy-releases: tried
[warn]   https://repo.typesafe.com/typesafe/ivy-releases/org.scalariform/scalariform_2.10/0.1.5-20140822-69e2e30/ivys/ivy.xml
[warn] ==== sbt-plugin-releases: tried
[warn]   https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scalariform/scalariform_2.10/0.1.5-20140822-69e2e30/ivys/ivy.xml
[warn] ==== local: tried
[warn]   /home/jenkins/.ivy2/local/org.scalariform/scalariform_2.10/0.1.5-20140822-69e2e30/ivys/ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/org/scalariform/scalariform_2.10/0.1.5-20140822-69e2e30/scalariform_2.10-0.1.5-20140822-69e2e30.pom
[warn] ==== bintray-eed3si9n-sbt-plugins: tried
[warn]   https://dl.bintray.com/eed3si9n/sbt-plugins/org.scalariform/scalariform_2.10/0.1.5-20140822-69e2e30/ivys/ivy.xml
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.scalariform#scalariform_2.10;0.1.5-20140822-69e2e30: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn] 	Note: Unresolved dependencies path:
[warn] 		org.scalariform:scalariform_2.10:0.1.5-20140822-69e2e30 (/scratch/jenkins/workspace/play-master-PRs/framework/project/plugins.sbt#L37-38)
```

**Comment by marcospereira** [comment]
@richdougherty it is clearer now. I will merge this pr when @play-pull-request-validator decides to play nice. :-)

**Comment by jroper** [comment]
Pushed 9b981e9 to solve this. @play-pull-request-validator

**Comment by richdougherty** [comment]
@play-pull-request-validator, please try again.

**Comment by richdougherty** [comment]
Resolution error resolved, now we get one of our standard transient errors:

```
[info] CompositeX509TrustManager should
[info]   with checkClientTrusted should
[error]     ! throws exception
[error]      unrecognized algorithm name: SHA256withRSA (AlgorithmId.java:407)
[error] sun.security.x509.AlgorithmId.get(AlgorithmId.java:407)
[error] sun.security.x509.X509CertImpl.sign(X509CertImpl.java:564)
[error] sun.security.x509.X509CertImpl.sign(X509CertImpl.java:528)
[error] play.api.libs.ws.ssl.CertificateGenerator$.generateCertificate(CertificateGenerator.scala:84)
```

@play-pull-request-validator, please rebuild again.

**Pull request merged**